### DISCUSSION TITLE
Connection string for on-prem and PowerBi Service

### DISCUSSION THREAD

**Discussion by Remko1981** [TabularEditor/TabularEditor3#1372] [🙋♀️TE3 Q&A] [answered]
Is there a provider which can be used in the connection string to connect to SQL server 2022 that works for on-prem SSAS models as well as Power Bi service semantic models?

We are planning on migrating our production SSAS models to the Power Bi Service. Our develop, test and acceptance models will remain on-prem. To facilitate this we need to change the provider in the connection string of the data source and the model property 'Default Power Bi Data Source Version'.

I tried the different options TE3 gives me. Of those only SQLOLEDB.1 works. But this has the draw back that all dates are recognized by the TE3 schema compare as strings.
I read that i could use System.Data.OleDb but i cannot find a way to install this so i can use it on-prem.

Any suggestions?

**Comment by otykier** [comment, marked as answer]
I believe the MSOLEDBSQL driver should work with both SSAS and Power BI models: https://learn.microsoft.com/en-us/sql/connect/oledb/oledb-driver-for-sql-server?view=sql-server-ver16#3-microsoft-ole-db-driver-for-sql-server-msoledbsql-recommended

Also, this should not have the drawback regarding date values, as you mention.

**Comment by Remko1981** [comment]
Thanks again Otykier! I do not know how i can have spend 2 hours on this and not see the error that i was using unsupported parameters... I had 'use encryption for data=True;' and 'trust server certificate=True' in the connection string.

The following connection string works: provider=MSOLEDBSQL;persist security info=False;initial catalog=<MyDatabase>;data source=<MyServer>
